tutorials/text/text_intro.py

```diff
@@ -93,3 +96,325 @@
+plt.show()
+
+##############################################################################
+# Vertical spacing for titles is controlled via ``rcParams[axes.titlepad]``,
+# which defaults to 5 points.  Setting to a different value moves tthe title.
+
+matplotlib.rcParams['axes.titlepad'] = 30.
+fig, ax = plt.subplots(figsize=(5, 3))
+fig.subplots_adjust(top=0.8)
+ax.plot(x1, y1)
+ax.set_title('Vertically offset title')
+plt.show()
+
+
+##############################################################################
+# Ticks and ticklabels
+# ====================
+#
+# Placing ticks and ticklabels is a very tricky aspect of making a figure.
+# Matplotlib does the best it can automatically, but it also offers a very
+# flexible framework for determining the choices for tick locations, and
+# how they are labelled.
+#
+# Terminology
+# ~~~~~~~~~~~
+#
+# *Axes*  (typically) have an `axis.xaxis` and `axis.yaxis` object that
+# contain the information about how the labels in the axis are laid out.
+#
+# The axis API is explained in detail in the `axis_api` section.
+
+# An axis object has major and minor ticks.  The axis has a *major_locator*
+# *minor_locator* methods that use the data being plotted to determine
+# the location of major and minor ticks.  There are also *major_formatter*
+# and *minor_formatter* methods that format the tick labels.
+#
+# Simple ticks
+# ~~~~~~~~~~~~
+#
+# While "discouraged", if can be very convenient to simply define the
+# tick values, and sometimes the tick labels, overridibg the default
+# locators and formatters.   Note that the second plot has the
+# ticks we asked for, including ones that are well outside the automatic
+# view limits.
+
+fig, axs = plt.subplots(2, 1, figsize=(5, 3), tight_layout=True)
+axs[0].plot(x1, y1)
+axs[1].plot(x1, y1)
+axs[1].xaxis.set_ticks(np.arange(0., 8.1, 2.))
+plt.show()
+
+#############################################################################
+# We can of course fix this after the fact, but it does highlight a
+# weakness of hard-coding the ticks.  This example also changes the format
+# of the ticks:
+fig, axs = plt.subplots(2, 1, figsize=(5, 3), tight_layout=True)
+axs[0].plot(x1, y1)
+axs[1].plot(x1, y1)
+ticks = np.arange(0., 8.1, 2.)
+# list comprehension to get all tick labels...
+tickla = ['%1.2f'%tick for tick in ticks]
+axs[1].xaxis.set_ticks(ticks)
+axs[1].xaxis.set_ticklabels(tickla)
+axs[1].set_xlim(axs[0].get_xlim())
+plt.show()
+
+#############################################################################
+# Tick Locators and Formatters
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# Instead of making a list of all the tickalbels, we could have
+# used a `ticker.FormatStrFormatter` and passed it to the `axis.xaxis`
+
+fig, axs = plt.subplots(2, 1, figsize=(5, 3), tight_layout=True)
+axs[0].plot(x1, y1)
+axs[1].plot(x1, y1)
+ticks = np.arange(0., 8.1, 2.)
+# list comprehension to get all tick labels...
+formatter = matplotlib.ticker.FormatStrFormatter('%1.1f')
```

Comment:
I would rather push forward StrMethodFormatter (i.e. using str.format), as old-style format strings are, well, old-style.
```
ticker.StrMethodFormatter("{x:1.1f}")
```
In fact I tend to just pass a bound method to FuncFormatter:
```
ticker.FuncFormatter("{:1.1f}".format)
```

Reply:
Ha, ok.  I just suck at the new format.  I’ll change!

Reply:
tbh I don't really like `ticker.StrMethodFormatter("{x:1.1f}")` either (IMO FuncFormatter is really the clearest) but I'm likely not be the best judge for what's best in a tutorial!

Reply:
I give an example of func formatter below.  Your use above is clever, but a bit obscure ;-)